### PR TITLE
In-Tree to CSI Migration logic for PersistentVolume Attach/Detach

### PR DESCRIPTION
**Pull request by ddebroy**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR picks up from https://github.com/kubernetes/kubernetes/pull/69830 and https://github.com/kubernetes/kubernetes/pull/70590: It "sets the stage" for in-tree to CSI Migration. It introduces the main feature flag, as well as the GCE PD and EBS specific feature flag, the IsMigratedToCSI() Plugin interface method, as well as an implementation of the Migration for Attach/Detach for PersistentVolumes only.

Feature tracking issue: kubernetes/enhancements#625

/sig storage
/kind feature
/assign @jsafrane @saad-ali
/cc @msau42 @davidz627 @leakingtapan

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This PR picks up from https://github.com/kubernetes/kubernetes/pull/69830 and https://github.com/kubernetes/kubernetes/pull/70590 with the changes rebased to master. The CSI migration feature work is no longer intended to be WIP.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: vendor/github.com/kubernetes-csi/kubernetes-csi-migration-library/plugins/aws_ebs.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/api/core/v1"
+)
+
+const (
+	AWSEBSDriverName       = "com.amazon.aws.csi.ebs"
```
Comment (davidz627) [inline vendor/github.com/kubernetes-csi/kubernetes-csi-migration-library/plugins/aws_ebs.go]: I think this was updated, can you bump the version of the library

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 2]
```diff
@@ -1437,3 +1495,31 @@ func isDeviceOpened(deviceToDetach AttachedVolume, mounter mount.Interface) (boo
 	}
 	return deviceOpened, nil
 }
+
+func isMigrated(vpm *volume.VolumePluginMgr, spec *volume.Spec) bool {
```
Comment (davidz627) [inline pkg/volume/util/operationexecutor/operation_generator.go]: add TODO comment that this should contain logic for checking CSINodeInfo for Kubelet migration status

**Comment by davidz627** [comment]
mostly LGTM, please squash new commits you authored as much as possible :+1:
ping @jsafrane @saad-ali this is the real deal now, I believe this PR has already been reviewed and approved for the most part. We want to start merging into k/k HEAD now

**Comment by davidz627** [comment]
Also might be worth doing that renaming that was asked for in this PR:
https://github.com/davidz627/kubernetes/pull/2/commits

I believe we changed the names of "Migratable" "Migrated" and such.
https://github.com/davidz627/kubernetes/pull/2#discussion_r229102490

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 3]
```diff
@@ -296,11 +302,29 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
...
+		// The volume represented by this spec is CSI and thus should be migrated
+		attachableVolumePlugin, err = og.volumePluginMgr.FindAttachablePluginByName(csi.CSIPluginName)
+		if err != nil || attachableVolumePlugin == nil {
+			eventRecorderFunc(&err)
+			return volumetypes.GeneratedOperations{}, volumeToAttach.GenerateErrorDetailed("AttachVolume.FindAttachablePluginBySpec failed", err)
+		}
+
+		csiSpec, err := translateSpec(volumeToAttach.VolumeSpec)
+		if err != nil {
+			return volumetypes.GeneratedOperations{}, volumeToAttach.GenerateErrorDetailed("AttachVolume.TranslateSpec failed", err)
```
Comment (leakingtapan) [inline pkg/volume/util/operationexecutor/operation_generator.go]: Should we record the err here using `eventRecordFunc` when translation failed similar to the above?

**Review by leakingtapan — COMMENTED** [review]
Should we include the [changes for mount/umount](https://github.com/davidz627/kubernetes/pull/4) in this PR all together? Or I can rebase my change once this is merged in

**Comment by ddebroy** [comment]
@leakingtapan will it be possible to rebase https://github.com/davidz627/kubernetes/pull/4 once this is in and submit to k/k please?

**Comment by leakingtapan** [comment]
> @leakingtapan will it be possible to rebase [davidz627#4](https://github.com/davidz627/kubernetes/pull/4) once this is in and submit to k/k please?

Sure. I will happy to rebase

**Pull request closed without merging**